### ISSUE TITLE
`force:source:beta:push` error message is not sufficiently precise

### ISSUE THREAD

**Issue by jclark-dot-org** [forcedotcom/cli#1311]
<!--
NOTICE: GitHub is not a mechanism for receiving support under any agreement or SLA. If you require immediate assistance, please use official support channels.
-->

### Summary

The new [beta][1] `push` command does not clearly report which custom field of an object caused a push failure.  

[1]: https://github.com/forcedotcom/cli/issues/1258

### Steps To Reproduce:

In any SFDX project with custom fields; a formula field is easiest way to demo:
1. `sfdx force:source:tracking:clear      # required before using beta source tracking`
2. edit a custom formula field to include an unencoded ampersand in the `<formula>`
3. `sfdx force:source:beta:tracking:push -f`

### Expected result

Error message should indicate the offending custom field.

### Actual result

Error message only indicates the affected SObject.

### System Information

```
{
	"cliVersion": "sfdx-cli/7.127.0",
	"architecture": "darwin-x64",
	"nodeVersion": "node-v16.13.0",
	"pluginVersions": [
		"@oclif/plugin-autocomplete 0.3.0 (core)",
		"@oclif/plugin-commands 1.3.0 (core)",
		"@oclif/plugin-help 3.2.3 (core)",
		"@oclif/plugin-not-found 1.2.4 (core)",
		"@oclif/plugin-plugins 1.10.1 (core)",
		"@oclif/plugin-update 1.5.0 (core)",
		"@oclif/plugin-warn-if-update-available 1.7.0 (core)",
		"@oclif/plugin-which 1.0.3 (core)",
		"@salesforce/sfdx-plugin-lwc-test 0.1.7 (core)",
		"alias 1.1.22 (core)",
		"apex 0.3.0 (core)",
		"auth 1.7.5 (core)",
		"config 1.2.47 (core)",
		"custom-metadata 1.0.12 (core)",
		"data 0.6.4 (core)",
		"generator 1.2.1 (core)",
		"limits 1.2.3 (core)",
		"org 1.9.0 (core)",
		"salesforce-alm 53.4.2 (core)",
		"schema 1.0.10 (core)",
		"sfdx-cli 7.127.0 (core)",
		"sfdx-fmz-plugin 1.1.0 (1.1.0)",
		"sfpowerkit 3.3.4",
		"source 1.4.0 (core)",
		"telemetry 1.2.8 (core)",
		"templates 52.6.0 (core)",
		"trust 1.0.10 (core)",
		"user 1.5.2 (core)"
	],
	"osVersion": "Darwin 18.7.0"
}
```

### Additional information

The beta `push` command is catching XML errors which are not caught by the original-recipe `push` command.  This is probably a good thing, however, the ambiguity of the error message required a lot of time to be spent finding the issue.

I am working with a long-established DX project, which includes a custom field `Customer_Signed_By__c` on the Contract object.  The `<formula>` in the `.field-meta.xml` file looks like this:

```
<formula>IF(!ISBLANK(CustomerSignedId) ,
HYPERLINK( CustomerSignedId, IF(Is_Underage_Applicant__c , Guardian__r.First_Name__C & ' ' & Guardian__r.Last_Name__c,Account.FirstName & ' ' & Account.LastName)) , 
Applicant_Signed_Email__c )</formula>
```

Notice there are a total of 4 bare ampersands being used for string concatenation.  As this is an xml file, they should be properly encoded:

```
<formula>IF(!ISBLANK(CustomerSignedId) ,
HYPERLINK( CustomerSignedId, IF(Is_Underage_Applicant__c , Guardian__r.First_Name__C &amp; ' ' &amp; Guardian__r.Last_Name__c,Account.FirstName &amp; ' ' &amp; Account.LastName)) ,
Applicant_Signed_Email__c )</formula>
```

The unencoded version has been working for over 18 months, and has been successfully pushed to hundreds of scratch orgs (and deployed to sandboxes & prod via MDAPI deploy).  When pushing with beta `push`, the following error was reported:

```
Type   Name      Problem                                                                                                                                                                                                                              
─────  ────────  ──────────────────────────────────────────────────────────────────────────────────────────────────────                                                                                                                               
Error  Contract  Error parsing file: The entity name must immediately follow the '&' in the entity reference. (1563:91)                                                                                                                               
```

It is clear the issue is XML entity encoding, and that it is related in some way to Contract, but there is no indication that the issue is in a custom field.  To track it down, I first tested all `Contract.*` files:

```
find ./force-app/* -name 'Contract.*' -print0 | xargs -0 xmllint > /dev/null
```

When that failed, I tried all XML files under `objects/Contract`, which found the issue:

```
> find ./force-app/main/default/objects/Contract -name '*.xml' -print0 | xargs -0 xmllint > /dev/null
./force-app/main/default/objects/Contract/fields/Customer_Signed_By__c.field-meta.xml:6: parser error : xmlParseEntityRef: no name                                                                                                                    
INK( CustomerSignedId, IF(Is_Underage_Applicant__c , Guardian__r.First_Name__C &                                           
                                                                               ^                                           
./force-app/main/default/objects/Contract/fields/Customer_Signed_By__c.field-meta.xml:6: parser error : xmlParseEntityRef: no name                                                                                                                    
ustomerSignedId, IF(Is_Underage_Applicant__c , Guardian__r.First_Name__C & ' ' &                                           
                                                                               ^                                           
./force-app/main/default/objects/Contract/fields/Customer_Signed_By__c.field-meta.xml:6: parser error : xmlParseEntityRef: no name                                                                                                                    
, Guardian__r.First_Name__C & ' ' & Guardian__r.Last_Name__c,Account.FirstName &                                           
                                                                               ^                                           
./force-app/main/default/objects/Contract/fields/Customer_Signed_By__c.field-meta.xml:6: parser error : xmlParseEntityRef: no name                                                                                                                    
dian__r.First_Name__C & ' ' & Guardian__r.Last_Name__c,Account.FirstName & ' ' &                                           
                                                                               ^                                           
```

**Comment by mshanemc** [comment]
wow, what a great report!  thank you

**Comment by uip-robot-zz** [comment]
This issue has been linked to a new work item: [W-10271490](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000VoHDWYA3/view)

**Comment by mshanemc** [comment]
@jclark-dot-org we dug into this one and here's what we found:

For deploy/push, we have to roll all the fields up into the object.  Then the metadata API gives us the errors at the file level (which in this case is the custom object, not the field).  So there's not a good way to tell the user what field caused the error because **we don't know**.

We're also going to not do extra encoding anymore on the new push commands (they'll now be consistent across all deploy/retrieve/push/push) that was previously hiding that xml error.  We checked the retrieve output from creating a new formula field and it comes encoded.
